games/minimal/mods/basetools/init.lua

```diff
@@ -0,0 +1,295 @@
+		},
+	},
+})
+minetest.register_tool("basetools:axe_stone", {
+	description = "Stone Axe",
+	inventory_image = "basetools_stoneaxe.png",
+	tool_capabilities = {
+		max_drop_level=0,
+		groupcaps={
+			choppy={times={[2]=1.00, [3]=0.60}, uses=60, maxlevel=0},
+		},
+	},
+})
+minetest.register_tool("basetools:axe_steel", {
+	description = "Steel Axe",
+	inventory_image = "basetools_steelaxe.png",
+	tool_capabilities = {
+		max_drop_level=1,
+		groupcaps={
+			choppy={times={[1]=2.00, [2]=0.80, [3]=0.40}, uses=90, maxlevel=0},
+		},
+	},
+})
+
+--
+-- Shears (dig snappy)
+--
+
+minetest.register_tool("basetools:shears_wood", {
+	description = "Wooden Shears",
+	inventory_image = "basetools_woodshears.png",
+	tool_capabilities = {
+		max_drop_level=0,
+		groupcaps={
+			snappy={times={[3]=1.00}, uses=30, maxlevel=0},
+		},
+	},
+})
+minetest.register_tool("basetools:shears_stone", {
+	description = "Stone Shears",
+	inventory_image = "basetools_stoneshears.png",
+	tool_capabilities = {
+		max_drop_level=0,
+		groupcaps={
+			snappy={times={[2]=1.00, [3]=0.50}, uses=60, maxlevel=0},
+		},
+	},
+})
+minetest.register_tool("basetools:shears_steel", {
+	description = "Steel Shears",
+	inventory_image = "basetools_steelshears.png",
+	tool_capabilities = {
+		max_drop_level=1,
+		groupcaps={
+			snappy={times={[1]=1.00, [2]=0.50, [3]=0.25}, uses=90, maxlevel=0},
+		},
+	},
+})
+
+--
+-- Swords (deal damage)
+--
+
+minetest.register_tool("basetools:sword_wood", {
+	description = "Wooden Sword",
+	inventory_image = "basetools_woodsword.png",
+	tool_capabilities = {
+		full_punch_interval = 1.0,
+		damage_groups = {fleshy=2},
+	}
+})
+minetest.register_tool("basetools:sword_stone", {
+	description = "Stone Sword",
+	inventory_image = "basetools_stonesword.png",
+	tool_capabilities = {
+		full_punch_interval = 1.0,
+		max_drop_level=0,
+		damage_groups = {fleshy=4},
+	}
+})
+minetest.register_tool("basetools:sword_steel", {
+	description = "Steel Sword",
+	inventory_image = "basetools_steelsword.png",
+	tool_capabilities = {
+		full_punch_interval = 1.0,
+		max_drop_level=1,
+		damage_groups = {fleshy=6},
+	}
+})
+
+-- Fire/Ice sword: Deal damage to non-fleshy damage groups
+minetest.register_tool("basetools:sword_fire", {
+	description = "Fire Sword",
+	inventory_image = "basetools_firesword.png",
+	tool_capabilities = {
+		full_punch_interval = 1.0,
+		max_drop_level=0,
+		damage_groups = {icy=6},
+	}
+})
+minetest.register_tool("basetools:sword_ice", {
+	description = "Ice Sword",
+	inventory_image = "basetools_icesword.png",
+	tool_capabilities = {
+		full_punch_interval = 1.0,
+		max_drop_level=0,
+		damage_groups = {firy=6},
+	}
+})
+
+--
+-- Dagger: Low damage, fast punch interval
+--
+minetest.register_tool("basetools:dagger_steel", {
+	description = "Steel Dagger",
+	inventory_image = "basetools_steeldagger.png",
+	tool_capabilities = {
+		full_punch_interval = 0.5,
+		max_drop_level=0,
+		damage_groups = {fleshy=2},
+	}
+})
```

Comment:
While no huge issue, you could get rid of lots of redundant code by having a table with tool tiers and their stats and then looping over that for registration.

Reply:
This is subjective code style. Serves its purpose and tables add unnecessary effort for little benefit here.